geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/sortedset/ZAddExecutor.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.geode.redis.internal.netty.Coder;
+import org.apache.geode.redis.internal.netty.Command;
+import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
+
+public class ZAddExecutor extends AbstractExecutor {
+  private final ZAddExecutorState zAddExecutorState = new ZAddExecutorState();
+
+  @Override
+  public RedisResponse executeCommand(Command command, ExecutionHandlerContext context) {
+    zAddExecutorState.initialize();
+    RedisSortedSetCommands redisSortedSetCommands = context.getRedisSortedSetCommands();
+    List<byte[]> commandElements = command.getProcessedCommand();
+    Iterator<byte[]> commandIterator = commandElements.iterator();
+
+    skipCommandAndKey(commandIterator);
+
+    int optionsFoundCount = findAndValidateZAddOptions(command, commandIterator, zAddExecutorState);
+    if (zAddExecutorState.exceptionMessage != null) {
+      return RedisResponse.error(zAddExecutorState.exceptionMessage);
+    }
+
+    return RedisResponse
+        .integer(redisSortedSetCommands.zadd(command.getKey(),
+            new ArrayList<>(commandElements.subList(optionsFoundCount + 2, commandElements.size())),
+            makeOptions(zAddExecutorState)));
+  }
+
+  private void skipCommandAndKey(Iterator<byte[]> commandIterator) {
+    commandIterator.next();
+    commandIterator.next();
+  }
+
+  private int findAndValidateZAddOptions(Command command, Iterator<byte[]> commandIterator,
+      ZAddExecutorState executorState) {
+    boolean scoreFound = false;
+    int optionsFoundCount = 0;
+
+    while (commandIterator.hasNext() && !scoreFound) {
+      String subCommandString = Coder.bytesToString(commandIterator.next()).toLowerCase();
+      switch (subCommandString) {
+        case "nx":
+          executorState.nxFound = true;
+          optionsFoundCount++;
+          break;
+        case "xx":
+          executorState.xxFound = true;
+          optionsFoundCount++;
+          break;
+        default:
+          try {
+            Double.valueOf(subCommandString);
+          } catch (NumberFormatException nfe) {
+            executorState.exceptionMessage = ERROR_NOT_A_VALID_FLOAT;
+          }
+          scoreFound = true;
+      }
+    }
+    if ((command.getProcessedCommand().size() - optionsFoundCount - 2) % 2 != 0) {
+      executorState.exceptionMessage = ERROR_SYNTAX;
+    } else if (executorState.nxFound && executorState.xxFound) {
+      executorState.exceptionMessage = ERROR_INVALID_ZADD_OPTION_NX_XX;
+    }
+    return optionsFoundCount;
+  }
+
+  private ZAddOptions makeOptions(ZAddExecutorState executorState) {
+    ZAddOptions.Exists existsOption = ZAddOptions.Exists.NONE;
+
+    if (executorState.nxFound) {
+      existsOption = ZAddOptions.Exists.NX;
+    }
+    if (executorState.xxFound) {
+      existsOption = ZAddOptions.Exists.XX;
+    }
+    return new ZAddOptions(existsOption);
+  }
+
+  static class ZAddExecutorState {
+    public boolean nxFound = false, xxFound = false;
```

Comment:
Sorry, just noticed this. Minor code style point, but according to [the code style guide](https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations) linked from [the Geode wiki](https://cwiki.apache.org/confluence/display/GEODE/Code+Style+Guide), we shouldn't declare multiple variables on one line.

Reply:
Pushed up a fix. Since the ZAddExecutorState is always initialized on the first line of the Executor, I also ditched the initialization. I'm pretty sure the compiler would optimize that, but what the heck.

Reply:
(Too bad neither IntelliJ nor spotless catch that. Of course, it seems like IntelliJ and spotless disagree about a few things, too.)

Reply:
Yeah, I do wonder if it might be a good idea as a community to review/update what spotless checks for, and maybe agree on some default IntelliJ warnings that should always be on and always be fixed. A discussion for the dev list maybe.